### ISSUE TITLE
S3 Dataspace not working when this addon is present

### ISSUE THREAD

**Issue by paraita** [ow2-proactive/emailnotification-addons#2]
The programming project is using a VFS S3 provider for the dataspaces but once the release is generated, this addon is failing because it cannot resolve that VFS S3 dependency. So the dependency has been added to this project's build.gradle to make this project compile again. As a result, the dataspaces through S3 is not failing at runtime, in the task requesting files through the data management:

```
[2016-09-06 13:04:05,938 ERROR           o.o.p.s.u.TaskLogger] task 1t0 (Linux_Bash_Task) error
java.lang.IllegalStateException: Invalid empty size list when trying to mount [s3://dstestparaita, s3://dstestparaita] mounted map content is {}
    at org.objectweb.proactive.extensions.dataspaces.vfs.VFSSpacesMountManagerImpl.mountFirstAvailableFileSystem(VFSSpacesMountManagerImpl.java:319)
    at org.objectweb.proactive.extensions.dataspaces.vfs.VFSSpacesMountManagerImpl.ensureVirtualSpaceIsMounted(VFSSpacesMountManagerImpl.java:269)
    at org.objectweb.proactive.extensions.dataspaces.vfs.VFSSpacesMountManagerImpl.resolveFile(VFSSpacesMountManagerImpl.java:168)
    at org.objectweb.proactive.extensions.dataspaces.vfs.VFSSpacesMountManagerImpl.resolveFile(VFSSpacesMountManagerImpl.java:138)
    at org.objectweb.proactive.extensions.dataspaces.core.DataSpacesImpl.resolveInputOutput(DataSpacesImpl.java:214)
    at org.objectweb.proactive.extensions.dataspaces.api.PADataSpaces.resolveOutput(PADataSpaces.java:485)
    at org.ow2.proactive.scheduler.task.data.TaskProActiveDataspaces$4.call(TaskProActiveDataspaces.java:255)
    at org.ow2.proactive.scheduler.task.data.TaskProActiveDataspaces$4.call(TaskProActiveDataspaces.java:252)
    at org.ow2.proactive.scheduler.task.data.TaskProActiveDataspaces.initDataSpace(TaskProActiveDataspaces.java:271)
    at org.ow2.proactive.scheduler.task.data.TaskProActiveDataspaces.initDataSpaces(TaskProActiveDataspaces.java:252)
    at org.ow2.proactive.scheduler.task.data.TaskProActiveDataspaces.<init>(TaskProActiveDataspaces.java:121)
    at org.ow2.proactive.scheduler.task.ProActiveForkedTaskLauncherFactory.createTaskDataspaces(ProActiveForkedTaskLauncherFactory.java:51)
    at org.ow2.proactive.scheduler.task.TaskLauncher.doTask(TaskLauncher.java:151)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at org.objectweb.proactive.core.mop.MethodCall.execute(MethodCall.java:353)
    at org.objectweb.proactive.core.body.request.RequestImpl.serveInternal(RequestImpl.java:214)
    at org.objectweb.proactive.core.body.request.RequestImpl.serve(RequestImpl.java:160)
    at org.objectweb.proactive.core.body.BodyImpl$ActiveLocalBodyStrategy.serveInternal(BodyImpl.java:552)
    at org.objectweb.proactive.core.body.BodyImpl$ActiveLocalBodyStrategy.serve(BodyImpl.java:485)
    at org.objectweb.proactive.core.body.AbstractBody.serve(AbstractBody.java:426)
    at org.objectweb.proactive.Service.serve(Service.java:128)
    at org.objectweb.proactive.Service.blockingServeOldest(Service.java:214)
    at org.objectweb.proactive.Service.blockingServeOldest(Service.java:189)
    at org.objectweb.proactive.Service.fifoServing(Service.java:150)
    at org.objectweb.proactive.core.body.ActiveBody$FIFORunActive.runActivity(ActiveBody.java:337)
    at org.objectweb.proactive.core.body.ActiveBody.run(ActiveBody.java:175)
    at java.lang.Thread.run(Thread.java:745)
[2016-09-06 13:04:05,939 INFO            o.o.p.s.u.TaskLogger] task 1t0 (Linux_Bash_Task) finished with errors
```

Removing this addon's jar from the addons folder solve the issue.
Four other repositories are also complaining about resolving this vfs-s3 dependency:
- connector-matlab-scilab
- connector-r
- nodesource-addons
- scheduling

**Comment by laurentpellegrino** [comment]
@paraita What about excluding _org.ow2.proactive:scheduler-api_ dependency from the generated JAR? This dependency is required for compiling and using the project without ProActive Scheduling. However, this dependency is already contained by the PWS distribution and thus not required in the generated project JAR. Excluding it should prevent to get transitive dependencies such as S3.

I have applied what I explained in the recent _inmemory-keyvalue-store-addons_ and it works perfectly: 

https://github.com/ow2-proactive/inmemory-keyvalue-store-addons/blob/master/build.gradle#L41

Such a change should work at least for _emailnotification-addons_ and will also decrease the size of the JAR since lot of duplicated dependencies will not be included.
